test/end-to-end/auth.js

```diff
@@ -213,3 +221,226 @@ describe("Login endpoint test", () => {
+                .send({ email: test_account.email });
+
+            expect(token.generateToken).toHaveBeenCalledWith({ email: test_account.email }, env.awt_secret, RECOVERY_LINK_EXPIRATION);
+
+            const generatedToken = generateTokenSpy.mock.results[0].value;
+            expect(res.status).toBe(HTTPStatus.OK);
+
+
+            const realTime = Date.now;
+            const mockDate = Date.now() + (RECOVERY_LINK_EXPIRATION * SECOND_IN_MS);
+            Date.now = () => mockDate;
+
+            res = await request()
+                .post(`/auth/recover/${generatedToken}/confirm`)
+                .send({ password: newPassword });
+
+            expect(res.status).toBe(HTTPStatus.FORBIDDEN);
+            expect(res.body).toHaveProperty("errors");
+            expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.INVALID_TOKEN);
+
+            Date.now = realTime;
+        });
+
+        describe("password", () => {
+            let generatedToken;
+
+            beforeAll(async () => {
+                await request()
+                    .post("/auth/recover/request")
+                    .send({ email: test_account.email });
+
+                expect(token.generateToken).toHaveBeenCalledWith({ email: test_account.email }, env.awt_secret, RECOVERY_LINK_EXPIRATION);
+
+                generatedToken = generateTokenSpy.mock.results[0].value;
+            });
+
+            const EndpointValidatorTester =
+                ValidatorTester((params) => request().post(`/auth/recover/${generatedToken}/confirm`).send(params));
+            const BodyValidatorTester = EndpointValidatorTester("body");
+            const FieldValidatorTester = BodyValidatorTester("password");
+            FieldValidatorTester.isRequired();
+            FieldValidatorTester.mustBeString();
+            FieldValidatorTester.hasMinLength(AccountConstants.password.min_length);
+            FieldValidatorTester.hasNumber();
+        });
+
+        test("recover flow", async () => {
+            const test_agent = agent();
+            let res = await test_agent
+                .post("/auth/login")
+                .send(test_account);
+
+            expect(res.status).toBe(HTTPStatus.OK);
+
+            await test_agent.delete("/auth/login");
+
+            res = await request()
+                .post("/auth/recover/request")
+                .send({ email: test_account.email });
+
+            expect(res.status).toBe(HTTPStatus.OK);
+            const generatedToken = generateTokenSpy.mock.results[0].value;
+
+            res = await request()
+                .post(`/auth/recover/${generatedToken}/confirm`)
+                .send({ password: newPassword });
+
+            expect(res.status).toBe(HTTPStatus.OK);
+
+            res = await test_agent
+                .post("/auth/login")
+                .send({ email: test_account.email, password: newPassword });
+
+            expect(res.status).toBe(HTTPStatus.OK);
```

Comment:
We could also check if the password hash effectively changed in the DB. What do you think?

Reply:
I think we could do that. It's being tested in the last test, but sure